### PR TITLE
added angularJS and removed jQuery

### PR DESCRIPTION
**Pull request by ytv**
Because I removed jQuery and added AngularJS, I had to rewrite the countdown which was previously a jquery plugin.  Also, the class names were slightly renamed to be more in line with CSS recommended style-guides.  The JS logic however is pretty much the same as before.

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -0,0 +1,8 @@
+RewriteEngine On
```
Comment (darrenvng) [inline .htaccess:1]: Why do you have an htaccess file, is this online?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# pomodoro-clock-angular
```
Comment (darrenvng) [inline README.md:1]: Nice that you are using font-awesome icons, for a different project try using google material icons to familiarize yourself with them.  Also, you don't need to mention javascript, html and css, its already implied.

File: main.html [hunk 3]
```diff
@@ -1,70 +1,63 @@
...
-                    <h1>Pomodoro Clock</h1>
-                </div>
-            </div>
-        </div>
-        <div class="col-md-4 col-md-offset-1 col-xs-12 text-center ctl">
-            <div class="col-md-12 col-xs-6 box">
-                <div>
-                    <h2>Break Length</h2>
+<html ng-app="app">
+    <head>
```
Comment (darrenvng) [inline main.html:3]: I like the meta viewport inclusion, test your app on your phone to make sure it really is responsive.  Try checking on your phone and ipad size device to get a feel for how it looks on mobile.

File: main.html [hunk 4]
```diff
@@ -1,70 +1,63 @@
...
+        <meta name="viewport" content="width = device-width, initial-scale = 1">
+        <link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.1/css/font-awesome.min.css">
+        <link rel="stylesheet" href="public/css/app.css">
+        <script src = "https://ajax.googleapis.com/ajax/libs/angularjs/1.4.9/angular.min.js"></script>
+        <script src="public/js/app.js"></script>
+        <title>Pomodoro Clock</title>
+    </head>
+    <body>
```
Comment (darrenvng) [inline main.html]: You should follow a style guide for angular js, this is a decent one https://github.com/johnpapa/angular-styleguide.  Names shouldn't be too specific because things always change, but they shouldn't be too general then its unclear.  If 'ctrl' is the main controller for your app, I would call it 'AppCtrl' (some ppls like 'AppController' but its too long for me). .What is 'box--ctrl'?  If text-center just applies text-align: center which centers everything, you can just call it center.  Maybe put it at the very top if most of your html uses centering?

File: main.html [hunk 5]
```diff
@@ -1,70 +1,63 @@
...
+        <div class="container" ng-controller='ctrl'>
+            <h1 class="text-center">Pomodoro Clock</h1>
+            <div class="col-xs-12 text-center box">
+                <div class="col-xs-6 box--ctrl">
+                    <h2>{{ labels.break }} Length</h2>
+                    <div>
+                        <span>
+                            <button type="button" class="btn"
```
Comment (darrenvng) [inline main.html]: Put your ng-click logic in a function, and call that function when user clicks.  It has 2 benefits:
1)  Your logic could grow later on, would be a hassle to change it in the html, especially if u have same logic in multiple spots.
2)  Code reuse.  If its a function, you can use it anywhere within the controller's scope, so you can replace all lines with this similar logic with this function.

File: public/js/app.js [hunk 6]
```diff
@@ -1,106 +1,82 @@
-$(function() {
-    Timer.init();
+var app = angular.module('app', []);
```
Comment (darrenvng) [inline public/js/app.js]: Angular files are generally structured in a specific way, check that link I showed you above.  Also generally you would put your app, ctrls and services in different files.  Its a little overkill for a project of this size, but then again u want to show you know these things.

File: public/js/app.js [hunk 7]
```diff
@@ -1,106 +1,82 @@
...
-            Timer._setCountdown();
-        }, Timer.timeRemaining);
-    },
-    _reset: function() {
-        $('#circleLabel').html('Start');
-        clearTimeout(Timer.timeTillNext);
-        if(Timer.paused === false)
-            $('#timer').countdown('stop');
-        $('#timer').html('00:00:00');
-        Timer.paused = true;
```
Comment (darrenvng) [inline public/js/app.js:104]: Btw, angular allows you to scale many similar items with ease.  Say I want 3 different clocks a the same time that function independently of each other?  With small changes, you can include a button to add another clock.  Use ng-repeat for this.  Basically restructure your app so that your main app controller keeps track of $scope.clocks = [{<length and labels>}, {<length and labels>}...].  Move the logic in here to another controller file called maybe ClockCtrl.  Then you should be able to leverage ng-repeat in the html to repeat the number of clocks.  You can test by manually changing $scope.clocks.  You can even add a delete button for each clock too.  Hint: each ctrl has its own scope but they all have access to the parent scope.  Use firebug to console.log $scope.

File: public/js/app.js [hunk 8]
```diff
@@ -1,106 +1,82 @@
...
+    // Re-initializes variables for the next countdown
+    $scope.setUpNextCountdown = function(length, label) {
+        $scope.timer = length;
+        $scope.timerLabel = label;
+        $scope.countdown();
+        bell.play();
+        paused = false;
+    };
+
+    // Function for the actual countdown
```
Comment (darrenvng) [inline public/js/app.js]: Prob shouldn't use the same 'countdown' name for your scope variable and local variable.

**Comment by darrenvng** [comment]
Always keep this in mind for front end, try to structure your clock (view) as a module of code (js/html/css) so that another developer can easily locate its whereabouts in js/html/css to make changes and also to make it scale.  Whatever view you create, I should be able to move it anywhere I want, add or remove more of them.  It won't always make sense for your practice projects, but u want to hit the recruiter in the face that without a doubt you understand the importance of making your front end easily scalable and maintainable. 

```
On Sunday, May 8, 2016 6:01 PM, ytv <notifications@github.com> wrote:
```

 Because I removed jQuery and added AngularJS, I had to rewrite the countdown which was previously a jquery plugin. Also, the class names were slightly renamed to be more in line with CSS recommended style-guides. The JS logic however is pretty much the same as before.
You can view, comment on, or merge this pull request online at:
  https://github.com/ytv/pomodoro-clock/pull/1
Commit Summary
- added angularJS and removed jQuery

File Changes
-  A .htaccess (8) 
-  A README.md (4) 
-  D README.me (3) 
-  M main.html (117) 
-  M public/css/app.css (27) 
-  M public/js/app.js (182) 

Patch Links:
- https://github.com/ytv/pomodoro-clock/pull/1.patch
- https://github.com/ytv/pomodoro-clock/pull/1.diff

File: .htaccess (hunk 1 above)
Comment (ytv) [inline .htaccess:1]: Yeah, you can pull it up on yingvuong.com

File: main.html (hunk 3 above)
Comment (ytv) [inline main.html:3]: Good idea.  It doesn't fit the dimensions on my phone.  Will fix.

File: main.html (hunk 5 above)
Comment (ytv) [inline main.html]: Ok, I created a decrement and increment function that ng-click will call upon

File: public/js/app.js (hunk 6 above)
Comment (ytv) [inline public/js/app.js]: I separated the javascript files by components.

File: public/js/app.js (hunk 8 above)
Comment (ytv) [inline public/js/app.js]: The countdown() function has been renamed to renderCountdown()

File: public/js/app.js (hunk 7 above)
Comment (ytv) [inline public/js/app.js:104]: Added functionality of adding additional timers, each with the option of muting/removing

File: main.html (hunk 4 above)
Comment (ytv) [inline main.html]: you're right, box--ctrl was confusing and didn't follow the style guide.  i renamed it box__content.

Also renamed the main controller 'AppCtrl" and moved up the text-align
